import_type_extraction/module_type_generator.py

```diff
@@ -0,0 +1,303 @@
+    #             type_data = (
+    #                 project['author'],
+    #                 project['repo'],
+    #                 file['filename'],
+    #                 file['types'],
+    #                 file['functions']
+    #             )
+
+    #             import_types.append(type_data)
+
+    #     if len(import_types) == 0:
+    #         print("Skipped...")
+    #         return
+        
+    #     type_df = pd.DataFrame(import_types, columns=columns)
+    #     type_df.to_csv(self.get_project_filename(project), index=False)
+
+    #     # Delete dataframe after writing to CSV to eventually let the garbage collector
+    #     # free memory.
+    #     del type_df
+
+
+    def filter_member(self, member_string: str, prefixes: list) -> bool:
+        """
+        Filters the specified member string with the given list of prefixes.
+        Returns false if the member string should be filtered out, and true
+        otherwise (if it should be kept)
+
+        :param: member_string  String representation of a member
+        :param: prefixes     List of prefixes (as strings) to omit
+        :return: True if member should be kept, False if it should be filtered out
+        """
+
+        # Check for empty string (can be obtained after trimming quotes)
+        if (len(member_string) == 0):
+            return False
+
+        for prefix in prefixes:
+            if (member_string.startswith(prefix)):
+                return False
+        
+        return True
+
+    def filter_members(self, members: list, prefixes: list) -> list:
+        """
+        Filters the specified list of members with the given list of prefixes.
+        Omits all members from members list that start with any string in prefixes.
+
+        :param: members   List of member strings (as list of strings)
+        :param: prefixes List of prefixes to use for filtering members out
+        """
+
+        return [t.strip('\'"') for t in members if self.filter_member(t.strip('\'"'), prefixes)]
+
+    def string_to_list(self, list_string: str) -> list:
+        """
+        Converts a Python string representation of a list into a Python list of strings.
+
+        :param: list_string  String representation of a list
+        :return: List of strings
+        """
+
+        return list_string.strip(']["\'').split(', ')
+
+    def concatenate_dataframes(self, out_dir: str, filename: str, prefix_filters = []) -> None:
+        """
+        Concatenates the CSV files present in the 'output_dir' location,
+        and saves them to a single CSV file.
+
+        :param: out_dir  Location to output CSV to (including filename)
+        :param: prefix_filters - List of prefixes to filter out for resulting members in final CSV.
+        """
+        # Create missing dirs if needed
+        os.makedirs(out_dir, exist_ok=True)
+
+        DATA_FILES = list_files(self.output_dir)
+        df = parse_df(DATA_FILES, batch_size=128)
```

Comment:
TODO: This should be done in chunks to accommodate for dataframes not fitting in memory